addons/binding/org.openhab.binding.hpprinter/src/main/java/org/openhab/binding/hpprinter/internal/handler/DefaultRequestHandler.java

```diff
@@ -0,0 +1,221 @@
+    private final String endpointUsage = "/DevMgmt/ProductUsageDyn.xml";
+    private final String endpointRestart = "/ProductActions/PowerCycle";
+
+    DefaultRequestHandler(HPPrinterConfiguration config, Thing thing) {
+        super(config, thing);
+        logger.debug("Using SVN_IPG_LEDM_119 parser for {}", thing.getUID().toString());
+    }
+
+    @Override
+    public void close() throws Exception {
+    }
+
+    @Override
+    public boolean checkIfOnline() {
+        WebRequester onlineCheck = new WebRequester("/", config);
+        return onlineCheck.check();
+    }
+
+    // Printer status table.
+    private String getPrinterStatusMessage(String statusMsg) {
+        Map<String, String> smap = new HashMap<>();
+
+        smap.put("processing", "Printing");
+        smap.put("inPowerSave", "Power Save");
+        smap.put("ready", "Idle");
+        smap.put("closeDoorOrCover", "Door Open");
+        smap.put("inkSystemInitializing", "Loading Ink");
+
+        return smap.getOrDefault(statusMsg, statusMsg);
+    }
+
+    @Override
+    public void checkStatus() {
+        WebRequester status = new WebRequester(endpointStatus, config);
+        logger.debug("Checking {} status", thing.getUID().toString());
+        WebRequesterChecker result = status.process();
+        switch (result.status()) {
+            case WebRequester.REQ_STATUS_OK:
+                NodeList statuses = result.document().getDocumentElement().getElementsByTagName("psdyn:Status");
+
+                String printerStatus = "";
+                for (int i = 0; i < statuses.getLength(); i++) {
+                    Element currInk = (Element) statuses.item(i);
+                    if (currInk.getElementsByTagName("pscat:StatusCategory").item(0).getTextContent() != "genuineHP") {
+                        printerStatus = getPrinterStatusMessage(
+                                currInk.getElementsByTagName("pscat:StatusCategory").item(0).getTextContent());
+                    }
+                }
+
+                updateState(new ChannelUID(thing.getUID(), CGROUP_STATUS, CHANNEL_STATUS),
+                        new StringType(printerStatus));
+
+                logger.debug("Check {} status ok", thing.getUID().toString());
+                break;
+
+            case WebRequester.REQ_STATUS_ENDPOINTFAIL:
+                goneOffline(true);
+                break;
+
+            case WebRequester.REQ_STATUS_DOCEXCEPTION:
+                logger.debug("Check {} status failed", thing.getUID().toString());
+                break;
+        }
+    }
+
+    @Override
+    public void checkUsage() {
+        WebRequester status = new WebRequester(endpointUsage, config);
+        logger.debug("Checking {} usage", thing.getUID().toString());
+        WebRequesterChecker result = status.process();
+        switch (result.status()) {
+            case WebRequester.REQ_STATUS_OK:
+                // Get Ink Levels
```

Comment:
Please don't make the body of switch longer dan approx 5 lines. So separate this body in its own method.